src/composables/composablesHelpers.ts

```diff
@@ -28,31 +33,46 @@ interface ICreateStorageRefOptions<T> {
  */
 export function useStorageRef<T = string | object | any[]>(
   initialState: T,
-  keys: string | string[],
-  { serializer }: ICreateStorageRefOptions<T> = {},
+  keys: StorageKeysInput,
+  options: ICreateStorageRefOptions<T> = {},
 ) {
+  const {
+    serializer,
+    backgroundSync = false,
+  } = options;
+
   let isRestored = false;
+  let watcherDisabled = false; // Avoid watcher going infinite loop
   const state = ref(initialState) as Ref<T>; // https://github.com/vuejs/core/issues/2136
-  const storageKeys: string[] = [...Array.isArray(keys) ? keys : [keys]].filter(excludeFalsy);
 
-  watch(state, (val) => {
-    WalletStorage.set(storageKeys, serializer?.write(val) || val);
+  function setState(val: any) {
+    if (val) {
+      watcherDisabled = true;
+      state.value = (serializer?.read) ? serializer.read(val) : val;
+      setTimeout(() => { watcherDisabled = false; }, 0);
+    }
+  }
+
+  watch(state, (val, oldVal) => {
+    // Arrays are not compared as there is a bug which makes the new and old val always the same.
```

Comment:
did you find the cause of the bug? Are you planning to resolve this bug?

Reply:
Unfortunately no. I know that this bug was present in Vue 2 but I thought in Vue 3 it was fixed. There's this note in the Vue documentation:
![image](https://github.com/aeternity/superhero-wallet/assets/8217670/a7c9bb9d-3c36-4a1b-bc1d-3798e64c3050)
I think this is the reason, but I'm not sure why this is happening only with the arrays here.
I talked about this with @lukeromanowicz but he also has no idea how to solve the problem.

Reply:
> Are you planning to resolve this bug?
This is not a bug on our side but it's a Vue issue. So for now I don't know how to approach it in a different way as I did.

Reply:
well... I guess we could do some hacking with cloning an array instead of pushing to force the ref change and maybe store it in second ref to have two-step update. But it's a dirty trick :/